src/include/simeng/OS/SimOS.hh

```diff
@@ -48,24 +49,45 @@ class SimOS {
  public:
   /** Construct a SimOS object. */
   SimOS(std::string executablePath, std::vector<std::string> executableArgs,
-        std::shared_ptr<simeng::memory::Mem> mem, bool setProcess = false);
+        std::shared_ptr<simeng::memory::Mem> mem);
 
   /** Tick SimOS. */
   void tick();
 
-  /** Get a process with specified TID. */
-  const std::shared_ptr<Process>& getProcess(uint64_t TID);
+  /** Create a new Process Object.
+   * A span<char> is optionally passed into the function to indicate if the
+   * Process is created via raw bytes or via a compiled binary.
+   * Returns the tid of the process that was created. */
+  uint64_t createProcess(std::optional<span<char>> instructionBytes = {});
 
-  /** Get shared_ptr to syscallHandler instance. */
-  std::shared_ptr<SyscallHandler> getSyscallHandler() const {
-    return syscallHandler_;
-  };
+  /** Get a process with specified `tid`. */
+  const std::shared_ptr<Process>& getProcess(uint64_t tid);
+
+  /** Terminate the process with threadID = `tid`, and set corresponding core
+   * to idle state if applicable. */
+  void terminateThread(uint64_t tid);
+
+  /** Terminate all processes with threadGroupID = `tgid`, and set corresponding
+   * cores to idle state if applicable. */
```

Comment:
*set the corresponding cores to an idle state

Reply:
Done